### PR TITLE
STYLE: Using f-strings in pandas/__init__.py

### PR DESCRIPTION
**Pull request by naomi172839**
Replaced the old style strings with the newer f-strings.

(Sorry for the confusion with the earlier draft)

#29886

### CODE REVIEW

**Comment by jreback** [comment]
lgtm ping on green.

File: pandas/__init__.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
     try:
         __import__(dependency)
     except ImportError as e:
-        missing_dependencies.append("{0}: {1}".format(dependency, str(e)))
+        missing_dependencies.append(f"{dependency}: {str(e)}")
```
Comment (alimcmaster1) [inline pandas/__init__.py]: Nitpick but no need for the str(e) here can be just `f"{dependency}: {e}"`

Comment (naomi172839) [reply pandas/__init__.py]: I was trying to stay consistent with the original code.  I figured that the original author, who is far more experienced than me, had a good reason for using `str(e)`.

Comment (alimcmaster1) [reply pandas/__init__.py]: f strings use __str__ by default?

**Comment by datapythonista** [comment]
Thanks @naomi172839

**Pull request merged**